code/game/objects/items/extinguisher.dm

```diff
@@ -45,6 +46,22 @@
 	create_reagents(max_water)
 	reagents.add_reagent("water", max_water)
 
+
+/obj/item/extinguisher/advanced
+	name = "advanced fire extinguisher"
+	desc = "Used to stop thermonuclear fires from spreading inside your engine."
+	icon_state = "foam_extinguisher0"
+	item_state = "foam_extimguisher1"
+	dog_fashion = null
+	tanktype = /obj/structure/reagent_dispensers/foamtank
+	precision = 1
+
+
+/obj/item/extinguisher/advanced/New()
+	..()
+	create_reagents(max_water)
+	reagents.add_reagent("firefighting_foam", max_water)

```

Comment:
You can probably make this a lot less snowflakey by making the extinguisher's reagents a var.

Also why the ***fuck*** is this using New() instead of Initialize()

Reply:
it didnt compile with a var

Reply:
What was the error? It's likely you were just using improper syntax.

Reply:
it had a var chem = water and in new it had reagent.add(chem,max)